okapi-core/src/main/java/org/folio/okapi/service/impl/ProcessModuleHandle.java

```diff
@@ -72,6 +73,9 @@ public ProcessModuleHandle(Vertx vertx, LaunchDescriptor desc, String id,
 
   @Override
   public Future<Void> start() {
+    if (process != null) {
+      return Future.failedFuture("already started " + commandLine);
+    }
     if (port == 0) {
       return start2();
     }
```

Comment:
A few lines below we have this code:

      socket.close();
      return Future.failedFuture(messages.getMessage("11502", Integer.toString(port)));

This doesn't wait until the socket has been closed. Better:

      return socket.close().otherwiseEmpty()
          .compose(x -> Future.failedFuture(messages.getMessage("11502", Integer.toString(port))));

Reply:
ok.